ui/src/views/CubeProjectEdit.vue

```diff
@@ -61,7 +62,8 @@ export default class CubeProjectEditView extends Vue {
       this.shape = await api.fetchOperationShape(operation)
     }
 
-    this.resource = Object.freeze(this.project?.pointer) ?? null
+    const namespace = this.project.csvMapping?.namespace ?? ''
+    this.resource = Object.freeze(this.project.pointer.addOut(cc.namespace, namespace))
```

Comment:
I'd rather we could avoid this kind of tweaks.

Reply:
I understand and I don't like it either, but having a nested form in the UI is not a good option either. There are multiple solutions to this problem:
- Have a separate screen to edit Mapping settings
- Magically provide `namespace` on the Project from the API
- Support complex sh:path in shaperone (e.g. `sh:path (cc:csvMapping cc:namespace)`)

For the disappearing namespace, I didn't analyze in detail, but I'm pretty sure it depends on if the full csvMapping is present in the dataset of the clownface pointer used in the form or not. In some cases, `project.csvMapping` is just a `link`. I think it worked by chance when first loading the form.

Reply:
No good options. Fair, enough, let's have it like this for now